### ISSUE TITLE
DEPRECATED mosquittopp?

### ISSUE THREAD

**Issue by Radagan** [eclipse/mosquitto#1386]
Hi, I was just about to implement a C++ client using mosquittopp and noticed that it's flagged for deprecation. Is this correct or am I just following outdated documentation on how to go about using mosquitto in C++? 

I'm really, really hoping you're going to continue supporting the wrapper. There aren't a lot of good options for MQTT C++ libraries on Linux/arm64 that I can find.

**Comment by hannes09** [comment]
Hi, I am a bit puzzled by this. Is mosquittopp deprecated?

**Comment by DasRoteSkelett** [comment]
Hi! 

I am also a little confused, I was also not able to find what Interface to use instead...

Can someone clarify?

**Comment by arcticskew** [comment]
Found this thread from when the change was made: https://github.com/eclipse/mosquitto/commit/f42a72b46baeedd906421a30f61519c945c518a9#comments

This mailing list thread has some more information on the different libraries and their support: https://www.eclipse.org/lists/mosquitto-dev/msg02084.html

**Comment by ncoghlan** [comment]
After evaluating both it and the Paho C++ library, we recently adopted `mosquittopp` into an existing in-house C++ asynchronous programming framework.

We looked at paho first, as we already used the paho Python libraries, but the problem we found with the Paho C++ libraries is that the abstraction choices in that library didn't play nice with the existing abstractions in our C++ code, and the two systems ended up fighting against each other, resulting in a confusing mess.

By dropping down to the lower level mosquittopp wrappers, we were able to build an interface that let mosquitopp do the heavy lifting for the actual MQTT support, while also integrating nicely into the rest of the system.

I suspect we won't be alone in that, with the higher level Paho interface being a better choice for green-field projects and systems, but the more C-like mosquittopp wrapper being easier to integrate into established C++ projects that already have their own notions of how network IO should be managed that don't align with the way the Paho library works.
